src/pages/Oauth/helpers.ts

```diff
@@ -0,0 +1,67 @@
+import { LoginFlow } from '@kybernetwork/oauth2'
+
+export const getSupportLoginMethods = (loginFlow: LoginFlow | undefined) => {
+  return loginFlow?.oauth_client?.metadata?.allowed_login_methods ?? []
+}
+
+const whiteListDomains = [/https:\/\/(.+?\.)?kyberswap\.com$/, /https:\/\/(.+)\.kyberengineering\.io$/]
+const isValidRedirectURL = (url: string | undefined) => {
+  try {
+    if (!url) return false
+    const newUrl = new URL(url) // valid url
+    if (
+      url.endsWith('.js') ||
+      newUrl.pathname.endsWith('.js') ||
+      !whiteListDomains.some(regex => newUrl.origin.match(regex))
+    ) {
+      return false
+    }
+    return newUrl.protocol === 'http:' || newUrl.protocol === 'https:'
+  } catch (error) {
+    return false
+  }
+}
+
+export const navigateToUrl = (url: string | undefined) => {
+  if (url && isValidRedirectURL(url)) window.location.href = url
```

Comment:
@viet-nv @XiaoYhun please help review this file carefully

Reply:
what about <a href="" /> @nguyenhoaidanh? 
can you help to test this if href is xss code?
If it can't execute xss when using `a` tag, I prefer switch to `a` tag (validate URL is still needed)

Reply:
updated:
- all a tag to ExternalLink
- validate URL in ExternalLink